match.go

```diff
@@ -280,9 +166,8 @@ func (c *Client) glob(dir, pattern string, matches []string) (m []string, e erro
 	return
 }
 
-// Join joins any number of path elements into a single path, adding
-// a Separator if necessary.
-// all empty strings are ignored.
+// Join joins any number of path elements into a single path, separating
```

Comment:
Like with `Match` I think we could import the godoc from `path.Join` directly.

Reply:
Done. However... we could also replace this by: "Join is a synonym for path.Join. Deprecated: use path.Join instead." (And similarly for Match, Split.) I'm not sure how much sense it makes to keep these synonyms around, given that even the rest of the package uses path functions directly. Only Glob is different.

Reply:
I think the functions were imported so that users could reliably use the ones in the package, rather than having to import Yet Another Package, which can even sometimes run into issues by accidentally mixing/matching one for the other.

It would be easy for someone to accidentally pick up `filepath` instead of `path` and not notice until someone on Windows tries to use their code.

So, I don’t think it’s necessary to deprecate these, or direct people to other options. But saying something like “this is a convenience wrapper around path.Join, so that you don’t have to import that library with this one. For details see that package.”

Reply:
Ok. Instead of, or in addition to, the descriptions from path?

Reply:
Any choice is pretty valid here.

Reply:
I've trimmed the descriptions down to their first sentence, followed by a reference to stdlib documentation at golang.org.